src/view/CookThisOrReRollView.java

```diff
@@ -127,6 +168,25 @@ private void updateView(CookThisOrReRollState cookThisOrReRollState) {
         servingsLabel.setText("Servings: " + cookThisOrReRollState.getServings());
         readyInMinutesLabel.setText("Ready In Minutes: " + cookThisOrReRollState.getReadyInMinutes());
         summaryTextArea.setText("Summary: " + cookThisOrReRollState.getSummary());
+
+        // Update favourite button icon
+        CookThisOrReRollState state = cookThisOrReRollViewModel.getState();
+        if (state.getIsFavourite()) {
+//            favouriteButton.setIcon(favouriteIconOn);
+            System.out.println("On");
+            JOptionPane.showMessageDialog(null,
+                    "This recipe was favourited",
+                    "Recipe favourited",
+                    JOptionPane.INFORMATION_MESSAGE);
+        } else {
+//            favouriteButton.setIcon(favouriteIconOff);
+            System.out.println("Off");
+            JOptionPane.showMessageDialog(null,
+                    "This recipe was unfavourited",
+                    "Recipe unfavourited",
+                    JOptionPane.INFORMATION_MESSAGE);
+        }
+        //titleImagePanel, otherComponentsPanel
```

Comment:
-Good you have 2 favourited or unfavourited

Reply:
I will clean up the code in next few commits and make a new pull request with testing 👍